### PR TITLE
GED: Improving CI

### PR DESCRIPTION
**Pull request by garomonegro**
Added generation as part of CI
Added golangci-lint

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    steps:
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.17
+      - uses: actions/checkout@v3
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          skip-go-installation: true
+          version: latest
```
Comment (danielhelfand) [inline .github/workflows/golangci-lint.yml:26]: We should probably pin to a specific version. Using latest will cause unexpected failures during pr submission that a submitter should not have to solve.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,36 +1,25 @@
-name: Test
-
+name: CI
 on:
   push:
-    branches: [ master ]
+    tags:
+      - v*
+    branches:
+      - master
```
Comment (danielhelfand) [inline .github/workflows/ci.yml:7]: Should we allow CI to run when not submitted to master?

File: .github/workflows/golangci-lint.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+name: golangci-lint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
```
Comment (danielhelfand) [inline .github/workflows/golangci-lint.yml:7]: Should we allow CI to run when not submitted to master?

File: .github/workflows/golangci-lint.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+          # Optional: if set to true then the all caching functionality will be complete disabled,
+          #           takes precedence over all other caching options.
+          # skip-cache: true
+
+          # Optional: if set to true then the action don't cache or restore ~/go/pkg.
+          # skip-pkg-cache: true
+
+          # Optional: if set to true then the action don't cache or restore ~/.cache/go-build.
+          # skip-build-cache: true
```
Comment (danielhelfand) [inline .github/workflows/golangci-lint.yml:48]: Should we remove comments? Maybe have a comment linking to general docs for golangci-lint action?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (garomonegro) [reply .github/workflows/ci.yml:7]: The CI is allow for all PR, regardless of the branch --- that is how its running on this PR --- but only on pushes to master branch.

File: .github/workflows/golangci-lint.yml (hunk 3 above)
Comment (garomonegro) [reply .github/workflows/golangci-lint.yml:7]: Same as above.

**Pull request merged**